unified-runtime/source/adapters/level_zero/v2/command_list_manager.cpp

```diff
@@ -160,11 +162,30 @@ ur_result_t ur_command_list_manager::appendRegionCopyUnlocked(
 wait_list_view ur_command_list_manager::getWaitListView(
     const ur_event_handle_t *phWaitEvents, uint32_t numWaitEvents,
     ur_event_handle_t additionalWaitEvent) {
-
+  uint32_t numWaitEventsEnabled = 0;
+  if (isImmediateCommandList) {
+    for (uint32_t i = 0; i < numWaitEvents; i++) {
```

Comment:
We should avoid doing that on each getWaitListView - this might impact performance. I'm wondering if it wouldn't be better to just always use regular events for command buffers... We wouldn't need that isInUse() workaround at all then.

Reply:
The feature this PR is adding isn't being exposed in SYCL-Graph right now, always using regular events rather than counter based events for command-buffers would compromise the performance of applications for all SYCL-Graph usage today.